File: pool_vd/src/core/models/PoolState/Vector2i.ts
export interface IVector2i {
  x: number;
  y: number;
}

export class Vector2i implements IVector2i {
  x = 0;
  y = 0;

  constructor(x: number, y: number) {
    this.x = x;
    this.y = y;
  }

  public add(v: Vector2i) {
    return new Vector2i(this.x + v.x, this.y + v.y);
  }

  public multiply(a: number | Vector2i) {
    if (a instanceof Vector2i) {
      return new Vector2i(this.x * a.x, this.y * a.y);
    }
    return new Vector2i(this.x * a, this.y * a);
  }

  public length() {
    return Math.sqrt(this.x ** 2 + this.y ** 2);
  }

  public clone() {
    return new Vector2i(this.x, this.y);
  }
}

File: video_processor/src/pool_state/Vector2i.py
from __future__ import annotations
import math

class Vector2i:
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y
  
    def add(self, v: Vector2i):
        return Vector2i(self.x + v.x, self.y + v.y)

    def multiply(self, a: int | Vector2i):
        if isinstance(a, Vector2i):
            return Vector2i(self.x * a.x, self.y * a.y)
        return Vector2i(self.x * a, self.y * a)

    def length(self):
        return math.sqrt(self.x ** 2 + self.y ** 2)